roofit/roofitcore/inc/RooMinimizer.h

```diff
@@ -44,26 +44,49 @@ class RooMinimizer : public TObject {
 public:
    enum class FcnMode { classic, gradient, generic_wrapper };
 
+   /// Config argument to RooMinimizer ctor
+   struct Config {
+      double recoverFromNaN = 10.; // RooAbsMinimizerFcn config
+      int optConst = 0;            // RooAbsMinimizerFcn config
+      int printEvalErrors = 10;    // RooAbsMinimizerFcn config
+      int doEEWall = 1;            // RooAbsMinimizerFcn config
+      int offsetting = -1;         // RooAbsMinimizerFcn config
+      const char *logf = nullptr;  // RooAbsMinimizerFcn config
+      int nWorkers = getDefaultWorkers(); // RooAbsMinimizerFcn config that can only be set in ctor
+      int parallelGradient = 0;          // RooAbsMinimizerFcn config that can only be set in ctor
+      int parallelLikelihood = 0;        // RooAbsMinimizerFcn config that can only be set in ctor
+      int verbose = 0;                    // local config
+      bool profile;                       // local config
+      std::string minimizerType = "";     // local config
+      FcnMode fcnMode = FcnMode::classic; // local config
+   private:
+      int getDefaultWorkers();
+   };
+
    explicit RooMinimizer(RooAbsReal &function, FcnMode fcnMode = FcnMode::classic);
-   explicit RooMinimizer(
-      std::shared_ptr<RooFit::TestStatistics::RooAbsL> likelihood,
-      RooFit::TestStatistics::LikelihoodMode likelihoodMode = RooFit::TestStatistics::LikelihoodMode::serial,
-      RooFit::TestStatistics::LikelihoodGradientMode likelihoodGradientMode =
-         RooFit::TestStatistics::LikelihoodGradientMode::multiprocess);
+   explicit RooMinimizer(RooAbsReal &function, Config &cfg);
```

Comment:
The input configuration struct should not be modified:

```
explicit RooMinimizer(RooAbsReal &function, Config const& cfg={});
```

Also, by default it should take a default-initialized configuration struct, such that the user doesn't have to pass a configuration struct explicitly if they only want to use the defaults anyway!

Also, why is this other constructor that Patrick introduced still needed now? I mean this one:
```C++
explicit RooMinimizer(RooAbsReal &function, FcnMode fcnMode = FcnMode::classic);
```

Your new constructor can also handle the "classic" non-new-style case now, right? So for a user (and to me), it's not clear which constructor to use when I have one of the old likelihoods.

Could you get rid of this other constructor, and then there is only one constructor, your new one with the optional configuration struct?

Reply:
Hi @guitargeek, sure, I actually thought this constructor was one from longer ago that I decided to keep for backwards compatibility.

`explicit RooMinimizer(RooAbsReal &function, FcnMode fcnMode = FcnMode::classic);`

Reply:
Ah, I see! It was only introduced in 6.26 for the multiprocessing interface, which is not expected to be much adopted by users so far. Meaning we still have the opportunity to get rid of it again and have one single constructor!